browser/js/products/products.js

```diff
@@ -0,0 +1,38 @@
+app.config(function ($stateProvider) {
+
+    $stateProvider.state('products', {
+        url: '/products',
+        controller: 'ProductsController',
+        templateUrl: 'js/products/products.html',
+        resolve: {
+          allProducts: function (ProductsFactory) {
+            return ProductsFactory.fetchAll();
+          }
+        }
+    });
+
+    $stateProvider.state('product', {
+        url: '/products/:productId',
```

Comment:
Is there any way to make this a child state? It seems silly to have what looks like a child state in terms of url not actually be a child state. My suggestion to do this would be making an empty `products` state and then defining a `products.list` and `products.single` state or something, where the list has an empty string as a url and the main `products` state has a template of just a `<ui-view></ui-view>`